readers/mutation_readers: queue_reader_handle_v2::push_end_of_stream() raise _ex if set

denesb opened the pull request:
Instead of raising std::runtime_error("Dangling queue_reader_handle_v2") unconditionally. push() already raises _ex if set, best to be consistent. Unconditionally raising std::runtime_error can cause an error to be logged, when aborting an operation involving a queue reader. Although the original exception passed to queue_reader_handle_v2::abort() is most likely handled by higher level code (not logged), the generic std::runtime_error raised is not and therefore is logged.

Fixes: https://github.com/scylladb/scylladb/issues/23550

Minor cosmetic bugfix, no backport required.

raphaelsc commented:
I guess you can even close https://github.com/scylladb/scylladb/issues/23550 with this one, since there's nothing left to do from the system perspective after your patch. QA should open an issue at their end. @cezarmoise

denesb commented:
> I guess you can even close #23550 with this one, since there's nothing left to do from the system perspective after your patch. QA should open an issue at their end. @cezarmoise

Good point, updated the commit with extra explanation on what problems this can cause and added Fixes.

raphaelsc commented on readers/mutation_readers.cc:
```diff
@@ -1358,18 +1368,15 @@ queue_reader_handle_v2& queue_reader_handle_v2::operator=(queue_reader_handle_v2
-        }
-        return make_exception_future<>(std::runtime_error("Dangling queue_reader_handle_v2"));
+    if (auto ex = check_abort(); ex) {
+        return make_exception_future<>(ex);
```
[[unlikely]] ?
